pkg/dogstatsd/parser.go

```diff
@@ -321,7 +324,15 @@ func parseMetricMessage(message []byte, namespace string, defaultHostname string
 
 	metricName := string(rawName)
 	if namespace != "" {
-		metricName = namespace + metricName
+		blacklisted := false
+		for _, prefix := range config.Datadog.GetStringSlice("statsd_metric_namespace_blacklist") {
```

Comment:
any sense of the perf impact of this code path? (we're in a hot path here)

Asking in particular because I believe calls to `config.Datadog` are relatively costly.

Also, I liked that this function didn't need `config.Datadog` :) But I don't feel strongly about it if you have good reasons to prefer this

Reply:
I removed the call to `config.Datadog`. The rest should be OK. Note that this code path is only used if the option is used.